ext/node/polyfills/_util/std_testing_diff.ts

```diff
@@ -234,16 +268,40 @@ export function diff<T>(A: T[], B: T[]): Array<DiffResult<T>> {
     );
   }
   return [
-    ...prefixCommon.map(
-      (c): DiffResult<typeof c> => ({ type: DiffType.common, value: c }),
+    ...new SafeArrayIterator(
+      ArrayPrototypeMap(
+        prefixCommon,
+        (c: T): DiffResult<typeof c> => ({ type: DiffType.common, value: c }),
+      ),
     ),
-    ...backTrace(A, B, fp[delta + offset], swapped),
-    ...suffixCommon.map(
-      (c): DiffResult<typeof c> => ({ type: DiffType.common, value: c }),
+    ...new SafeArrayIterator(backTrace(A, B, fp[delta + offset], swapped)),
+    ...new SafeArrayIterator(
+      ArrayPrototypeMap(
+        suffixCommon,
+        (c: T): DiffResult<typeof c> => ({ type: DiffType.common, value: c }),
+      ),
     ),
   ];
 }
 
+const ESCAPE_PATTERN = new SafeRegExp(/([\b\f\t\v])/g);
+const ESCAPE_MAP = ObjectFreeze({
+  "\b": "\\b",
+  "\f": "\\f",
+  "\t": "\\t",
+  "\v": "\\v",
+});
+const LINE_BREAK_GLOBAL_PATTERN = new SafeRegExp(/\r\n|\r|\n/g);
```

Comment:
I used this code as a reference

https://github.com/denoland/deno/blob/32438d25c337f9160c1c90d48680963654385e22/ext/console/01_console.js#L2540-L2548